File: postman/sdks/Specs SDKs/Search Service/typescript/src/services/search-endpoints/models/charts-default-model.ts
import { z } from 'zod';
import { GenreModel, genreModel, genreModelRequest, genreModelResponse } from './genre-model';

/**
 * Zod schema for the ChartsDefaultModel model.
 * Defines the structure and validation rules for this data type.
 * This is the shape used in application code - what developers interact with.
 */
export const chartsDefaultModel = z.lazy(() => {
  return z.object({
    score: z.number(),
    chartId: z.number(),
    chartName: z.string(),
    artistId: z.number().optional().nullable(),
    artistName: z.string().optional().nullable(),
    createDate: z.string(),
    isApproved: z.number(),
    updateDate: z.string(),
    enabled: z.number(),
    isIndexed: z.number(),
    personId: z.number().optional().nullable(),
    publishDate: z.string().optional().nullable(),
    itemTypeId: z.number().optional().nullable(),
    personUsername: z.string().optional().nullable(),
    isPublished: z.number(),
    trackCount: z.number().optional().nullable(),
    chartImageUri: z.string().optional().nullable(),
    chartImageDynamicUri: z.string().optional().nullable(),
    genres: z.array(genreModel).optional().nullable(),
  });
});

/**
 *
 * @typedef  {ChartsDefaultModel} chartsDefaultModel
 * @property {number}
 * @property {number}
 * @property {string}
 * @property {number}
 * @property {string}
 * @property {string}
 * @property {number}
 * @property {string}
 * @property {number}
 * @property {number}
 * @property {number}
 * @property {string}
 * @property {number}
 * @property {string}
 * @property {number}
 * @property {number}
 * @property {string}
 * @property {string}
 * @property {GenreModel[]}
 */
export type ChartsDefaultModel = z.infer<typeof chartsDefaultModel>;

/**
 * Zod schema for mapping API responses to the ChartsDefaultModel application shape.
 * Handles any property name transformations from the API schema.
 * If property names match the API schema exactly, this is identical to the application shape.
 */
export const chartsDefaultModelResponse = z.lazy(() => {
  return z
    .object({
      score: z.number(),
      chart_id: z.number(),
      chart_name: z.string(),
      artist_id: z.number().optional().nullable(),
      artist_name: z.string().optional().nullable(),
      create_date: z.string(),
      is_approved: z.number(),
      update_date: z.string(),
      enabled: z.number(),
      is_indexed: z.number(),
      person_id: z.number().optional().nullable(),
      publish_date: z.string().optional().nullable(),
      item_type_id: z.number().optional().nullable(),
      person_username: z.string().optional().nullable(),
      is_published: z.number(),
      track_count: z.number().optional().nullable(),
      chart_image_uri: z.string().optional().nullable(),
      chart_image_dynamic_uri: z.string().optional().nullable(),
      genres: z.array(genreModelResponse).optional().nullable(),
    })
    .transform((data) => ({
      score: data['score'],
      chartId: data['chart_id'],
      chartName: data['chart_name'],
      artistId: data['artist_id'],
      artistName: data['artist_name'],
      createDate: data['create_date'],
      isApproved: data['is_approved'],
      updateDate: data['update_date'],
      enabled: data['enabled'],
      isIndexed: data['is_indexed'],
      personId: data['person_id'],
      publishDate: data['publish_date'],
      itemTypeId: data['item_type_id'],
      personUsername: data['person_username'],
      isPublished: data['is_published'],
      trackCount: data['track_count'],
      chartImageUri: data['chart_image_uri'],
      chartImageDynamicUri: data['chart_image_dynamic_uri'],
      genres: data['genres'],
    }));
});

/**
 * Zod schema for mapping the ChartsDefaultModel application shape to API requests.
 * Handles any property name transformations required by the API schema.
 * If property names match the API schema exactly, this is identical to the application shape.
 */
export const chartsDefaultModelRequest = z.lazy(() => {
  return z
    .object({
      score: z.number(),
      chartId: z.number(),
      chartName: z.string(),
      artistId: z.number().optional().nullable(),
      artistName: z.string().optional().nullable(),
      createDate: z.string(),
      isApproved: z.number(),
      updateDate: z.string(),
      enabled: z.number(),
      isIndexed: z.number(),
      personId: z.number().optional().nullable(),
      publishDate: z.string().optional().nullable(),
      itemTypeId: z.number().optional().nullable(),
      personUsername: z.string().optional().nullable(),
      isPublished: z.number(),
      trackCount: z.number().optional().nullable(),
      chartImageUri: z.string().optional().nullable(),
      chartImageDynamicUri: z.string().optional().nullable(),
      genres: z.array(genreModelRequest).optional().nullable(),
    })
    .transform((data) => ({
      score: data['score'],
      chart_id: data['chartId'],
      chart_name: data['chartName'],
      artist_id: data['artistId'],
      artist_name: data['artistName'],
      create_date: data['createDate'],
      is_approved: data['isApproved'],
      update_date: data['updateDate'],
      enabled: data['enabled'],
      is_indexed: data['isIndexed'],
      person_id: data['personId'],
      publish_date: data['publishDate'],
      item_type_id: data['itemTypeId'],
      person_username: data['personUsername'],
      is_published: data['isPublished'],
      track_count: data['trackCount'],
      chart_image_uri: data['chartImageUri'],
      chart_image_dynamic_uri: data['chartImageDynamicUri'],
      genres: data['genres'],
    }));
});

File: postman/sdks/Specs SDKs/Search Service/python/src/search_service_sdk/models/charts_default_model.py
from __future__ import annotations
from typing import List
from pydantic import Field
from typing import Optional
from typing import Union
from .utils.base_model import BaseModel
from .genre_model import GenreModel


class ChartsDefaultModel(BaseModel):
    """ChartsDefaultModel

    :param score: score
    :type score: float
    :param chart_id: chart_id
    :type chart_id: int
    :param chart_name: chart_name
    :type chart_name: str
    :param artist_id: artist_id, defaults to None
    :type artist_id: int, optional
    :param artist_name: artist_name, defaults to None
    :type artist_name: str, optional
    :param create_date: create_date
    :type create_date: str
    :param is_approved: is_approved
    :type is_approved: int
    :param update_date: update_date
    :type update_date: str
    :param enabled: enabled
    :type enabled: int
    :param is_indexed: is_indexed
    :type is_indexed: int
    :param person_id: person_id, defaults to None
    :type person_id: int, optional
    :param publish_date: publish_date, defaults to None
    :type publish_date: str, optional
    :param item_type_id: item_type_id, defaults to None
    :type item_type_id: int, optional
    :param person_username: person_username, defaults to None
    :type person_username: str, optional
    :param is_published: is_published
    :type is_published: int
    :param track_count: track_count, defaults to None
    :type track_count: int, optional
    :param chart_image_uri: chart_image_uri, defaults to None
    :type chart_image_uri: str, optional
    :param chart_image_dynamic_uri: chart_image_dynamic_uri, defaults to None
    :type chart_image_dynamic_uri: str, optional
    :param genres: genres, defaults to None
    :type genres: List[GenreModel], optional
    """

    score: float
    chart_id: int
    chart_name: str
    artist_id: Optional[int] = Field(default=None)
    artist_name: Optional[str] = Field(default=None)
    create_date: str
    is_approved: int
    update_date: str
    enabled: int
    is_indexed: int
    person_id: Optional[int] = Field(default=None)
    publish_date: Optional[str] = Field(default=None)
    item_type_id: Optional[int] = Field(default=None)
    person_username: Optional[str] = Field(default=None)
    is_published: int
    track_count: Optional[int] = Field(default=None)
    chart_image_uri: Optional[str] = Field(default=None)
    chart_image_dynamic_uri: Optional[str] = Field(default=None)
    genres: Optional[List[GenreModel]] = Field(default=None)
